src/main/java/core/basesyntax/service/impl/ReturnTransactionHandler.java

```
@@ -0,0 +1,22 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.db.Database;
+import core.basesyntax.model.FruitType;
+import core.basesyntax.service.TransactionHandler;
+import java.math.BigInteger;
+
+public class ReturnTransactionHandler implements TransactionHandler {
+    private static final TransactionHandler INSTANCE = new ReturnTransactionHandler();
+
+    private ReturnTransactionHandler() {
+    }
+
+    public static TransactionHandler instance() {
+        return INSTANCE;
+    }
+
+    @Override
+    public void handle(Database database, FruitType fruitType, BigInteger amount) {
+        database.add(fruitType, amount);
```

This method lacks null-argument validation. The generalized requirements state: "Null handling: validate method arguments (e.g., file paths, lists) are not null and throw informative RuntimeException if violated." Add explicit checks for `database`, `fruitType`, and `amount` at the start of `handle` and throw informative RuntimeException messages (fail fast) instead of risking NPEs later.